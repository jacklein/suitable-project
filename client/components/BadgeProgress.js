import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';
import { Bar } from 'react-native-progress';
import { progressBar } from '../styles/common';

const calculatePercent = decimal => {
  return (100 * parseFloat(decimal)).toFixed(2);
}

const BadgeProgress = ({ badge }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        Badge Progress: {calculatePercent(badge.details.progress)}%
      </Text>
      <Bar 
        progress={badge.details.progress}
        color={progressBar.color}
        height={progressBar.height}
      />
    </View>
  )
};

const styles = StyleSheet.create({
  container: {
    flex: 1, 
    alignItems: 'center',
    marginBottom: 15
  },
  text: {
    fontSize: 16, 
    marginVertical: 10
  }
});

BadgeProgress.propTypes = {
  badge: PropTypes.object,
}


export default BadgeProgress