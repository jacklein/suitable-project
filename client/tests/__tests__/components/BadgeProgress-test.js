import React from 'react';
import { shallow } from 'enzyme';
import BadgeProgress from '../../../components/BadgeProgress';
import { achievements } from '../../mockdata';

describe('BadgeProgress component', () => {
  test('renders correctly', () => {
    const badge = achievements[0];
    const wrapper = shallow(
      <BadgeProgress progress={badge.details.progress} />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ progress: 0.99 });
    expect(wrapper).toMatchSnapshot();
  });
});