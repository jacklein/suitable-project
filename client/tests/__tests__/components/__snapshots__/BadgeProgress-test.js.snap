// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeProgress component renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 15,
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 16,
        "marginVertical": 10,
      }
    }
  >
    Badge Progress: 
    29.70
    %
  </Component>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={4}
    borderWidth={1}
    color="#4492E5"
    height={15}
    indeterminate={false}
    progress={0.29699748744715526}
    useNativeDriver={false}
    width={150}
  />
</Component>
`;

exports[`BadgeProgress component renders correctly 2`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 15,
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 16,
        "marginVertical": 10,
      }
    }
  >
    Badge Progress: 
    99.00
    %
  </Component>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={4}
    borderWidth={1}
    color="#4492E5"
    height={15}
    indeterminate={false}
    progress={0.99}
    useNativeDriver={false}
    width={150}
  />
</Component>
`;
